apps/hpc-ftsadmin/src/app/components/flow-form.tsx

```diff
@@ -0,0 +1,4094 @@
+            group: 'reportChannel',
+            id: item.reportChannel && item.reportChannel.value,
+            name: item.reportChannel && item.reportChannel.displayLabel,
+          },
+        };
+      }),
+      flowType: {
+        id: values.flowType && values.flowType.value,
+        name: values.flowType && values.flowType.displayLabel,
+        group: 'flowType',
+      },
+      keywords: values.keywords.map((item) => ({
+        id: item.value,
+        name: item.displayLabel,
+        group: 'keywords',
+      })),
+      flowStatuses: {
+        id: values.flowStatus && values.flowStatus.value,
+        name: values.flowStatus && values.flowStatus.displayLabel,
+        group: 'flowStatus',
+      },
+      contributionTypes: {
+        id: values.contributionType && values.contributionType.value,
+        name: values.contributionType && values.contributionType.displayLabel,
+        group: 'contributionType',
+      },
+      method: {
+        id: values.method && values.method.value,
+        name: values.method && values.method.displayLabel,
+        group: 'method',
+      },
+      earmarking: values.earmarkingType
+        ? {
+            id: values.earmarkingType.value,
+            name: values.earmarkingType.displayLabel,
+            group: 'earmarkingType',
+          }
+        : null,
+      categories: [] as (string | number)[],
+      isCancellation: isPending ? true : !isPending ? false : null,
+      cancelled: isPending && isCancellation ? true : null,
+      pendingStatus: isPending ? true : !isPending ? false : [],
+      planEntities: isPending ? true : !isPending ? false : [],
+      planIndicated: isPending ? true : !isPending ? false : [],
+      isApprovedFlowVersion:
+        rejectFlag && approveFlag
+          ? true
+          : !(rejectFlag && approveFlag)
+          ? false
+          : null,
+      inactiveReason: [
+        {
+          id: null as null | number | string,
+          name: '' as string | undefined,
+          description: null,
+          parentID: null,
+          code: null,
+          group: '',
+          includeTotals: null,
+          createdAt: '',
+          updatedAt: '',
+        },
+      ],
+      isErrorCorrection: errorCorrection
+        ? true
+        : isSuperseded
+        ? true
+        : isPending
+        ? true
+        : !isSuperseded && !isPending
+        ? false
+        : null,
+      rejected: rejectFlag ? true : !rejectFlag ? false : null,
+      versions:
+        versionData &&
+        versionData.map((item: VersionDataType) => {
+          const items = {
+            id: item.flowId,
+            versionID: item.versionId,
+            activeStatus: item.activeStatus,
+            isPending: item.isPending ? item.isPending : false,
+            isCancelled: item.isCancelled ? item.isCancelled : false,
+          };
+          return items;
+        }),
+      ...fundingObject,
+    };
+
+    data = collapseCategories(data);
+    return data;
```

Comment:
(Based on the comment above of making data a `const`)
```suggestion
    return collapseCategories(data);
```